ci/lava/tests/test-config-across-reboot.py

```diff
@@ -0,0 +1,190 @@
+                break
+            print("DUT is still offline. Trying again...")
+        assert dut_address is not None
+
+    def test_get_new_hostname(self, dut, dut_addr, execute_helper):
+        """Get the hostname from the DUT.
+
+        Check the new hostname matches the new version.
+        """
+        err, hostname = self._get_hostname(dut, dut_addr, execute_helper)
+
+        assert err == 0
+        assert hostname is not None
+        assert TestConfigAcrossReboot.newHostname is not None
+        assert hostname == TestConfigAcrossReboot.newHostname
+
+    def test_get_wlan_ip_after_reboot(self, execute_helper):
+        """Perform the test on the DUT via the mbl-cli."""
+        self._ifconfig(execute_helper)
+        err, wlan_ip_address = self._get_wlan_ip(execute_helper)
+        assert err == 0
+        assert wlan_ip_address is not None
+
+    def _randomStringDigits(self, stringLength=6):
+        """Generate a random string of letters and digits."""
+        lettersAndDigits = string.ascii_letters + string.digits
+        return "".join(
+            random.choice(lettersAndDigits) for i in range(stringLength)
+        )
+
+    def _get_hostname(self, dut, dut_addr, execute_helper):
+        """Perform the test on the DUT via the mbl-cli."""
+        hostname = None
+
+        err, stdout, stderr = execute_helper.send_mbl_cli_command(
+            ["shell", 'sh -l -c "hostname"'], dut_addr
+        )
+
+        lines = stdout.splitlines()
+
+        if dut == "auto":
+            if len(lines) >= 2:
+                hostname = lines[1]
+        else:
+            for line in lines:
+                if dut in line:
+                    hostname = line
+        if hostname:
+            print("_get_hostname: " + hostname)
+        else:
+            print("_get_hostname: No hostname found")
+
+        return err, hostname
+
+    def _get_wlan_ip(self, execute_helper):
+        """Perform the test on the DUT via the mbl-cli."""
+        wlan_ip_address = None
+
+        err, stdout, stderr = execute_helper.send_mbl_cli_command(
+            ["shell", 'sh -l -c "ifconfig wlan0"'],
+            TestConfigAcrossReboot.dut_address,
+        )
+
+        if err == 0:
+            for line in stdout.splitlines():
+                if re.search(r"^.*inet addr:(?:\d{1,3}\.){3}\d{1,3}.*", line):
+                    wlan_ip_address = line
+
+        return err, wlan_ip_address
+
+    def _disable_technology(self, technology, execute_helper):
+        self._ifconfig(execute_helper)
+        test_command = 'sh -l -c "connmanctl disable {}"'.format(technology)
+
+        err, stdout, stderr = execute_helper.send_mbl_cli_command(
+            ["shell", test_command], TestConfigAcrossReboot.dut_address
+        )
+
+        # Wait for the interface to be removed and the routing tables
+        # etc to be updated
+        time.sleep(30)
+        self._ifconfig(execute_helper)
```

Comment:
What is this for?

Reply:
This is debug code that will show the ifconfig of the DUT in the Lava Log - might be useful to understand the reason if the IP address part of the test fails.